src/controllers/api/friends.js

```diff
@@ -0,0 +1,41 @@
+const { User } = require("../../models");
+
+const createFriendForUser = async (req, res) => {
+  try {
+    const { userId } = req.params;
+    const newFriend = await User.findByIdAndUpdate(
+      userId,
+      {
+        $push: { friends: req.body._id },
```

Comment:
What if there is no `_id` field included in the body? Currently you have no error handling for an empty body or an incorrect body. Let's destructure the body on line 6, and then we can add an `if` block to make sure we have the correct keys included:

`const { _id } = req.body`
`if (_id) { ...... }` 

Only if the `_id` is present do we want to create a friend. Otherwise we should send back an error message to say that the ID number was missing.

Reply:
Done.
Thank you so much.